app/src/main/java/com/daniel/card_game_android/MainActivity.java

```diff
@@ -89,27 +91,28 @@ private void setNewCardsImage(String imageNameA, String imageNameB) {
 
     private void setScore(Card playerCardA, Card playerCardB) {
         if (playerCardA.isStronger(playerCardB)) {
-            playerScoreA++;
-            mainViewController.setPlayerScore(playerScoreA + "");
+            playerA.setPlayerScore(playerA.getPlayerScore() + 1);
+            mainViewController.setPlayerScore(playerA.getPlayerScore() + "");
         } else {
-            playerScoreB++;
-            mainViewController.setComputerScore(playerScoreB + "");
+            playerB.setPlayerScore(playerB.getPlayerScore() + 1);
+            mainViewController.setComputerScore(playerB.getPlayerScore() + "");
         }
     }
 
     private void setProgress() {
         double sizeOfBar = 100 / (NUMBER_OF_CARDS / 2.0);
-        double totalTurns = playerScoreA + playerScoreB;
+        double totalTurns = playerA.getPlayerScore() + playerB.getPlayerScore();
         int gameProgress = (int) (totalTurns * sizeOfBar);
         mainViewController.setProgressBar(gameProgress);
     }
 
     private void displayWinner() {
         Intent intent = new Intent(this, WinnerPage.class);
-        intent.putExtra(WinnerPage.playerScoreA, playerScoreA);
-        intent.putExtra(WinnerPage.playerScoreB, playerScoreB);
-        intent.putExtra(MainActivity.gander, playerGander);
-        intent.putExtra(name, playerName);
+        intent.putExtra(WinnerPage.playerImageA, playerA.getPlayerImage());
+        intent.putExtra(WinnerPage.playerScoreA, playerA.getPlayerScore());
+        intent.putExtra(WinnerPage.playerScoreB, playerB.getPlayerScore());
+        intent.putExtra(MainActivity.gender, playerA.getPlayerGender());
+        intent.putExtra(name, playerA.getPlayerName());
```

Comment:
We need to use json and deliver the player to the next page and not value by value